docs/src/rest/partition-spec.md

```diff
@@ -0,0 +1,161 @@
+## Lance Partition Spec (Experimental)
```

Comment:
One thought, can we make this a part of the DirectoryNamespace catalog spec, because it is basically a directory namespace + all tables must have the same schema, and the namespace must be leveled.